lib/std/meta.zig

```diff
@@ -691,12 +691,13 @@ test "std.meta.TagPayload" {
 }
 
 /// Compares two of any type for equality. Containers are compared on a field-by-field basis,
-/// where possible. Pointers are not followed.
+/// where possible, or using the `eql()` function, if defined. Pointers are not followed.
 pub fn eql(a: anytype, b: @TypeOf(a)) bool {
     const T = @TypeOf(a);
 
     switch (@typeInfo(T)) {
         .Struct => |info| {
+            if (comptime std.meta.trait.hasFn("eql")(@TypeOf(a))) return a.eql(b);
```

Comment:
This applies to other container types like unions or opaque ptrs, no?

Reply:
Oh, right!

 The test has been moved before the `switch`.

Thank you!